### ISSUE TITLE
../../.eslintrc.json#overrides[0]': Class extends value undefined is not a constructor or null

### ISSUE THREAD

**Issue by gagle** [angular-eslint/angular-eslint#843]
I still get the following error with latest versions of everything after updating Angular v13.

````
Failed to load plugin '@nrwl/nx' declared in 'apps/XXX/.eslintrc.json » ../../.eslintrc.json#overrides[0]': Class extends value undefined is not a constructor or null
````

**apps/XXX/.eslintrs.json**

````
{
  "extends": "../../.eslintrc.json",
  "overrides": [
    {
      "files": ["*.ts"],
      "rules": {
        ...
      }
    }
  ]
}
````

**root .eslintrc.json**

````
{
  "root": true,
  "plugins": [
    "@angular-eslint",
    "@nrwl/nx",
    "@typescript-eslint",
    "import",
    "no-use-extend-native",
    "prefer-arrow",
    "unicorn",
    "simple-import-sort",
    "unused-imports"
  ],
  "overrides": [
    {
      "files": ["*.ts"],
      "parserOptions": {
        "project": ["tsconfig.*?.json"],
        "createDefaultProgram": true
      },
      "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:@angular-eslint/recommended",
        "plugin:@nrwl/nx/typescript",
        "plugin:import/errors",
        "plugin:import/warnings",
        "plugin:unicorn/recommended",
        "plugin:compat/recommended",
        "plugin:prettier/recommended",
        "plugin:jest/recommended",
        "plugin:jest/style"
      ],
      "rules": {
        ...
      }
    },
    ...
  ]
}
````

````
"dependencies": {
    "@angular/animations": "13.0.2",
    "@angular/cdk": "13.0.2",
    "@angular/common": "13.0.2",
    "@angular/compiler": "13.0.2",
    "@angular/core": "13.0.2",
    "@angular/forms": "13.0.2",
    "@angular/material": "13.0.2",
    "@angular/platform-browser": "13.0.2",
    "@angular/platform-browser-dynamic": "13.0.2",
    "@angular/router": "13.0.2",
    "@datorama/akita": "6.2.4",
    "@datorama/akita-ng-router-store": "6.0.2",
    "@datorama/akita-ngdevtools": "6.0.1",
    "@ngneat/forms-manager": "2.5.0",
    "@ngrx/effects": "13.0.1",
    "@ngrx/router-store": "13.0.1",
    "@ngrx/store": "13.0.1",
    "@ngx-translate/core": "14.0.0",
    "@ngx-translate/http-loader": "7.0.0",
    "bootstrap": "4.6.1",
    "core-js": "3.19.1",
    "datebook": "7.0.7",
    "dayjs": "1.10.7",
    "ngx-quill": "16.0.1",
    "quill": "1.3.7",
    "rxjs": "7.4.0",
    "tslib": "2.3.1",
    "uuid": "8.3.2",
    "zone.js": "0.11.4"
  },
  "devDependencies": {
    "@angular-builders/custom-webpack": "13.0.0",
    "@angular-devkit/architect": "0.1300.3",
    "@angular-devkit/build-angular": "13.0.3",
    "@angular-devkit/core": "13.0.3",
    "@angular-devkit/schematics": "13.0.3",
    "@angular-eslint/builder": "13.0.1",
    "@angular-eslint/eslint-plugin": "13.0.1",
    "@angular-eslint/eslint-plugin-template": "13.0.1",
    "@angular-eslint/schematics": "13.0.1",
    "@angular-eslint/template-parser": "13.0.1",
    "@angular/cli": "13.0.3",
    "@angular/compiler-cli": "13.0.2",
    "@angular/language-service": "13.0.2",
    "@jest/console": "27.3.1",
    "@jest/core": "27.3.1",
    "@jest/environment": "27.3.1",
    "@jest/fake-timers": "27.3.1",
    "@jest/reporters": "27.3.1",
    "@jest/test-result": "27.3.1",
    "@ngneat/spectator": "9.0.0",
    "@ngrx/schematics": "13.0.1",
    "@ngrx/store-devtools": "13.0.1",
    "@nrwl/angular": "13.2.2",
    "@nrwl/cli": "13.2.2",
    "@nrwl/eslint-plugin-nx": "13.2.2",
    "@nrwl/jest": "13.2.2",
    "@nrwl/linter": "13.2.2",
    "@nrwl/nx-cloud": "12.5.4",
    "@nrwl/storybook": "13.2.2",
    "@nrwl/tao": "13.2.2",
    "@nrwl/workspace": "13.2.2",
    "@storybook/addon-actions": "6.3.0-alpha.29",
    "@storybook/addon-backgrounds": "6.3.0-alpha.29",
    "@storybook/addon-knobs": "6.3.0-alpha.21",
    "@storybook/angular": "6.3.0-alpha.29",
    "@storybook/builder-webpack5": "6.2.9",
    "@types/jest": "27.0.3",
    "@types/jest-when": "2.7.3",
    "@types/node": "16.11.9",
    "@types/uuid": "8.3.3",
    "@types/webpack": "5.28.0",
    "@typescript-eslint/eslint-plugin": "5.4.0",
    "@typescript-eslint/parser": "5.4.0",
    "canvas": "2.8.0",
    "cross-env": "7.0.3",
    "dotenv": "10.0.0",
    "eslint": "8.3.0",
    "eslint-config-prettier": "8.3.0",
    "eslint-import-resolver-typescript": "2.5.0",
    "eslint-plugin-compat": "4.0.0",
    "eslint-plugin-import": "2.25.3",
    "eslint-plugin-jest": "25.3.0",
    "eslint-plugin-no-use-extend-native": "0.5.0",
    "eslint-plugin-prefer-arrow": "1.2.3",
    "eslint-plugin-prettier": "4.0.0",
    "eslint-plugin-simple-import-sort": "7.0.0",
    "eslint-plugin-unicorn": "39.0.0",
    "eslint-plugin-unused-imports": "2.0.0",
    "husky": "7.0.4",
    "jasmine-marbles": "0.8.4",
    "jest": "27.3.1",
    "jest-canvas-mock": "2.3.1",
    "jest-extended": "1.1.0",
    "jest-html-reporters": "2.1.6",
    "jest-junit": "13.0.0",
    "jest-message-util": "27.3.1",
    "jest-preset-angular": "11.0.1",
    "jest-runner": "27.3.1",
    "jest-runtime": "27.3.1",
    "jest-sonar": "0.2.12",
    "jest-when": "3.4.2",
    "js-beautify": "1.14.0",
    "lint-staged": "12.1.2",
    "mkdirp": "1.0.4",
    "mockdate": "3.0.5",
    "ng-mocks": "12.5.0",
    "ngx-markdown": "13.0.0",
    "patch-package": "6.4.7",
    "prettier": "2.4.1",
    "prettier-eslint": "13.0.0",
    "rimraf": "3.0.2",
    "rxjs-marbles": "7.0.1",
    "source-map-explorer": "2.5.2",
    "stylelint": "14.1.0",
    "stylelint-config-idiomatic-order": "8.1.0",
    "stylelint-config-prettier": "9.0.3",
    "stylelint-config-sass-guidelines": "9.0.1",
    "stylelint-config-standard": "24.0.0",
    "stylelint-declaration-block-no-ignored-properties": "2.5.0",
    "stylelint-no-unsupported-browser-features": "5.0.2",
    "stylelint-order": "5.0.0",
    "stylelint-prettier": "2.0.0",
    "stylelint-scss": "4.0.0",
    "stylelint-use-nesting": "3.0.0",
    "svg-to-ts": "6.0.1",
    "timezone-mock": "1.2.2",
    "ts-jest": "27.0.7",
    "ts-node": "10.4.0",
    "type-fest": "1.2.2",
    "typescript": "4.4.4"
  }
````

**Comment by nzacca** [comment]
@gagle Not sure what happened but my response appears to have disappeared. Lets try again.

I believe we are waiting on the nx team to release a new version for these commits: https://github.com/nrwl/nx/compare/13.2.2...master

Specifically, we're waiting on this PR which adds eslint 8 support: https://github.com/nrwl/nx/pull/7800

Cheers!

**Comment by nzacca** [comment]
My mistake -- I just realized that your original comment was on another thread. :)

**Comment by raysuelzer** [comment]
Hmmm. I have this same error with v13.2 and the latest ESLint 8.2. Tried rebuilding but no dice. 

```
"dependencies": {
    "@angular/animations": "13.2.2",
    "@angular/cdk": "13.2.2",
    "@angular/cli": "^13.2.3",
    "@angular/common": "13.2.2",
    "@angular/compiler": "13.2.2",
    "@angular/core": "13.2.2",
    "@angular/forms": "13.2.2",
    "@angular/material": "13.2.2",
    "@angular/platform-browser": "13.2.2",
    "@angular/platform-browser-dynamic": "13.2.2",
    "@angular/router": "13.2.2",
    "@angular/service-worker": "13.2.2",
    "@apollo/client": "^3.0.0",
    "@kolkov/angular-editor": "^2.0.0",
    "@mapbox/mapbox-gl-draw": "^1.1.2",
    "@ng-select/ng-select": "^6.1.0",
    "@ngrx/effects": "^13.0.2",
    "@ngrx/router-store": "^13.0.2",
    "@ngrx/store": "^13.0.2",
    "@ngrx/store-devtools": "^13.0.2",
    "@sentry/browser": "^5.6.2",
    "@turf/points-within-polygon": "^5.1.5",
    "angular-token": "^7.0.1",
    "apollo-angular": "^2.3.0",
    "autolinker": "^3.14.1",
    "core-js": "3.2.1",
    "date-fns": "^2.28.0",
    "date-fns-tz": "^1.2.2",
    "flag-icon-css": "^3.5.0",
    "graphql": "^15.0.0",
    "immer": "^4.0.0",
    "immutability-helper": "3.0.1",
    "jquery": "3.4.1",
    "libphonenumber-js": "^1.9.16",
    "lodash-es": "^4.17.15",
    "mapbox-gl": "~2.4.1",
    "ngx-mapbox-gl": "~6.0.4",
    "ngx-virtual-scroller": "^4.0.3",
    "papaparse": "^5.0.2",
    "qrcode": "^1.4.4",
    "rxjs": "6.5.5",
    "stream": "0.0.2",
    "text-mask-addons": "3.8.0",
    "tslib": "^2.0.0",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~13.2.3",
    "@angular-eslint/builder": "^12.0.0",
    "@angular-eslint/eslint-plugin": "^12.0.0",
    "@angular-eslint/eslint-plugin-template": "^12.0.0",
    "@angular-eslint/schematics": "13.1.0",
    "@angular-eslint/template-parser": "^12.0.0",
    "@angular/compiler-cli": "13.2.2",
    "@angular/language-service": "13.2.2",
    "@graphql-codegen/add": "^2.0.2",
    "@graphql-codegen/cli": "^1.20.1",
    "@graphql-codegen/fragment-matcher": "^2.0.1",
    "@graphql-codegen/typescript": "^1.21.0",
    "@graphql-codegen/typescript-apollo-angular": "^2.3.0",
    "@graphql-codegen/typescript-compatibility": "^2.0.1",
    "@graphql-codegen/typescript-document-nodes": "^1.17.9",
    "@graphql-codegen/typescript-operations": "^1.17.3",
    "@types/jasmine": "~3.6.0",
    "@types/jasminewd2": "~2.0.6",
    "@types/jquery": "3.3.31",
    "@types/lodash-es": "^4.17.3",
    "@types/mapbox-gl": "^0.54.3",
    "@types/node": "^12.11.1",
    "@types/node-fetch": "^2.5.0",
    "@types/toastr": "^2.1.37",
    "@typescript-eslint/eslint-plugin": "5.3.0",
    "@typescript-eslint/parser": "5.3.0",
    "aws-sdk": "^2.518.0",
    "eslint": "^8.2.0",
    "eslint-plugin-import": "2.25.2",
    "eslint-plugin-jsdoc": "30.7.6",
    "eslint-plugin-prefer-arrow": "1.2.2",
    "fs": "0.0.1-security",
    "git-branch-is": "3.0.0",
    "glob": "^7.1.4",
    "gql2ts": "~1.10.1",
    "humps": "^2.0.1",
    "jasmine-core": "~3.6.0",
    "jasmine-spec-reporter": "~5.0.0",
    "jscodeshift": "^0.11.0",
    "karma": "~6.3.9",
    "karma-chrome-launcher": "~3.1.0",
    "karma-cli": "~2.0.0",
    "karma-coverage-istanbul-reporter": "~3.0.2",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "^1.5.0",
    "node-fetch": "^2.6.0",
    "path": "^0.12.7",
    "protractor": "~7.0.0",
    "randomstring": "^1.1.5",
    "requestretry": "^4.0.0",
    "ts-node": "8.3.0",
    "typescript": "4.5.5",
    "webpack-bundle-analyzer": "3.4.1",
    "webpack-sources": "1.4.3"
  }
```

.eslintrc.json 

```{
  "root": true,
  "ignorePatterns": [
    "projects/**/*"
  ],
  "parserOptions": {
    "ecmaVersion": 11
  },
  "env": {
    "es2020": true
  },
  "overrides": [
    {
      "files": [
        "*.ts"
      ],
      "parserOptions": {
        "project": [
          "tsconfig.json",
          "e2e/tsconfig.e2e.json"
        ],
        "createDefaultProgram": true
      },
      "extends": [
        "plugin:@angular-eslint/ng-cli-compat",
        "plugin:@angular-eslint/ng-cli-compat--formatting-add-on",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "rules": {
        "prefer-arrow/prefer-arrow-functions": "off",
        "@angular-eslint/component-class-suffix": [
          "error",
          {
            "suffixes": [
              "Component",
              "Page",
              "Dialog"
            ]
          }
        ],
        "@angular-eslint/component-selector": [
          "error",
          {
            "type": "element",
            "prefix": "app",
            "style": "kebab-case"
          }
        ],
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "app",
            "style": "camelCase"
          }
        ],
        "@typescript-eslint/naming-convention": [
          0
        ],
        "@typescript-eslint/consistent-type-definitions": "error",
        "@typescript-eslint/dot-notation": "off",
        "@typescript-eslint/explicit-member-accessibility": [
          "off",
          {
            "accessibility": "explicit"
          }
        ],
        "@typescript-eslint/no-unused-expressions": "off",
        "@typescript-eslint/member-ordering": [
          0,
          {
            "order": [
              "static-field",
              "instance-field",
              "static-method",
              "instance-method"
            ]
          }
        ],
        "@typescript-eslint/no-shadow": "error",
        "@typescript-eslint/indent": [
          "off",
          2
        ],
        "brace-style": [
          "error",
          "1tbs",
          {
            "allowSingleLine": true
          }
        ],
        "no-console": "warn",
        "id-blacklist": "off",
        "id-match": "off",
        "no-multiple-empty-lines": "error",
        "no-underscore-dangle": "off",
        "no-shadow": "off"
      }
    },
    {
      "files": [
        "*.html"
      ],
      "extends": [
        "plugin:@angular-eslint/template/recommended"
      ],
      "rules": {}
    }
  ]
}
```

**Comment by bsara** [comment]
@nzacca this issue has been fixed for v13, but v12 is still not compatible with ESLint 8 and I am getting this error. Will there be a release of v12 that is compatible with eslint 8+?

**Comment by nzacca** [comment]
@bsara As far as I know, there has been no backport of this change in nx v12. You'll likely need to update your nx version to v13 for eslint 8 support.

**Comment by bsara** [comment]
@nzacca I'm not using nx, I'm just using the eslint plugin.

**Comment by nzacca** [comment]
@bsara I'm not sure about that. This ticket pertains to nx itself. As far as v12 of this package (angular-eslint), I don't know if that will be updated to support eslint 8.

**Comment by JamesHenry** [comment]
Hi folks, I'm going to close this one as ESLint 8 has been supported in both nx and angular-eslint for quite a while now. If new issues arise please create a new issue with steps to reproduce for angular-eslint. Many thanks!

**Comment by nzacca** [comment]
@bsara This package `angular-eslint` is not geared toward a specific framework but the fix does need to be ported to a version of `angular-eslint` that supports Angular 12. As per my comment above, I don't know if that port will ever happen.

Based on the release notes [here](https://github.com/angular-eslint/angular-eslint/blob/master/CHANGELOG.md#1300-2021-11-18), version 8 of eslint is only supported by v13+ of `angular-eslint` which requires you to upgrade to Angular 13.

Cheers!

**Comment by bsara** [comment]
@nzacca thanks for the clarification.

> requires you to upgrade to Angular 13.

That's unfortunate for me I guess. We still need to support IE 11 😢
